gcore/resource_gcore_baremetal.go

```diff
@@ -89,7 +89,8 @@ func resourceBmInstance() *schema.Resource {
 			},
 			"name": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
+				Computed: true,
```

Comment:
make sense to add ExactOneOf to name and name_templates, 
```
ExactlyOneOf: []string{
    "name",
    "name_templates",
}
```
may be the same should be done with instance resource